pkg/testing/fixture_install.go

```diff
@@ -0,0 +1,128 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package testing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+
+	"github.com/elastic/elastic-agent/pkg/core/process"
+)
+
+type CmdOpts interface {
+	toCmdArgs() []string
+}
+
+type EnrollOpts struct {
+	URL             string // --url
+	EnrollmentToken string // --enrollment-token
+}
+
+func (e EnrollOpts) toCmdArgs() []string {
+	var args []string
+	if e.URL != "" {
+		args = append(args, "--url", e.URL)
+	}
+	if e.EnrollmentToken != "" {
+		args = append(args, "--enrollment-token", e.EnrollmentToken)
+	}
+	return args
+}
+
+type InstallOpts struct {
+	BasePath       string // --base-path
+	Force          bool   // --force
+	NonInteractive bool   // --non-interactive
+
+	EnrollOpts
+}
+
+func (i InstallOpts) toCmdArgs() []string {
+	var args []string
+	if i.BasePath != "" {
+		args = append(args, "--base-path", i.BasePath)
+	}
+	if i.Force {
+		args = append(args, "--force")
+	}
+	if i.NonInteractive {
+		args = append(args, "--non-interactive")
+	}
+
+	args = append(args, i.EnrollOpts.toCmdArgs()...)
+
+	return args
+}
+
+// Install installs the prepared Elastic Agent binary
+func (f *Fixture) Install(ctx context.Context, installOpts *InstallOpts, opts ...process.CmdOption) ([]byte, error) {
+	installArgs := []string{"install"}
+	installArgs = append(installArgs, installOpts.toCmdArgs()...)
+	out, err := f.Exec(ctx, installArgs, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	f.installed = true
+	f.installOpts = installOpts
+
+	f.t.Cleanup(func() {
+		_, err := f.Uninstall(ctx, nil)
+		require.NoError(f.t, err)
```

Comment:
I think we should check for `ErrNotInstalled` (see other comment). That way if the test code calls `Install()` followed by `Uninstall()` this cleanup will no a noop and won't error.